gradle/integration-test.gradle

```diff
@@ -0,0 +1,27 @@
+sourceSets {
+  integrationTest {
+    kotlin.srcDirs = ['src/integrationTest/kotlin']
+    resources.srcDir file('src/integrationTest/resources')
+    compileClasspath += sourceSets.main.output + configurations.testRuntime
+    runtimeClasspath += output + compileClasspath
+  }
+}
+
+configurations {
+  integrationTestCompile.extendsFrom testCompile
+  integrationTestRuntime.extendsFrom testRuntime
+}
+
+gradlePlugin {
```

Comment:
What is this one referring to?

Reply:
This is from the official [Gradle test kit docs](https://docs.gradle.org/current/userguide/test_kit.html#sub:test-kit-automatic-classpath-injection).

> The Java Gradle Plugin development plugin can be used to assist in the development of Gradle plugins. Starting with Gradle version 2.13, the plugin provides a direct integration with TestKit. **When applied to a project, the plugin automatically adds the gradleTestKit() dependency to the test compile configuration.** Furthermore, it automatically generates the classpath for the code under test and injects it via GradleRunner.withPluginClasspath() for any GradleRunner instance created by the user. It’s important to note that the mechanism currently only works if the plugin under test is applied using the plugins DSL. If the target Gradle version is prior to 2.8, automatic plugin classpath injection is not performed.

I need to double check though if I can avoid `gradleTestKit()` dependency since this plugin configuration was added here. 👍

Reply:
Thanks for raising this question, this construct is used to `reconfigure the conventions provided by the Java Gradle Plugin Development plugin for a project that uses a custom Test source set` which is not the case for this project and not needed.